File: packages/plugin-backfill/src/chunking/analyze.ts
import { generateChunkPlan } from './planner.js'
import type { ChunkPlan, GenerateChunkPlanInput } from './types.js'

export type AnalyzeAndChunkInput = GenerateChunkPlanInput
export type AnalyzeAndChunkResult = ChunkPlan
export type AnalyzeTableInput = GenerateChunkPlanInput
export type AnalyzeTableResult = ChunkPlan

export async function analyzeAndChunk(input: AnalyzeAndChunkInput): Promise<AnalyzeAndChunkResult> {
  return generateChunkPlan(input)
}

export async function analyzeTable(input: AnalyzeTableInput): Promise<AnalyzeTableResult> {
  return analyzeAndChunk(input)
}

File: chkit_python/src/chkit_plugin_backfill/chunking/analyze.py
"""Analyze facade — port of ``chunking/analyze.ts``."""

from __future__ import annotations

from chkit_plugin_backfill.chunking.planner import generate_chunk_plan
from chkit_plugin_backfill.chunking.types import ChunkPlan, GenerateChunkPlanInput


def analyze_and_chunk(input_: GenerateChunkPlanInput) -> ChunkPlan:
    return generate_chunk_plan(input_)


def analyze_table(input_: GenerateChunkPlanInput) -> ChunkPlan:
    return analyze_and_chunk(input_)


__all__ = ["analyze_and_chunk", "analyze_table"]
